### PR TITLE
fixed format warnings in release builds

### PR DESCRIPTION
**Pull request by fmoessbauer**
Fixes all `-Wformat` warnings in the tests. Someone owns me a beer for that.

### CODE REVIEW

File: dash/test/io/HDF5ArrayTest.cc [hunk 1]
```diff
@@ -568,10 +569,10 @@ TEST_F(HDF5ArrayTest, TeamSplit) {
     EXPECT_NE_U(array_a.lbegin(), nullptr);
 
     if (array_a.lbegin() != nullptr) {
-      LOG_MESSAGE("Team %d: read array", myteam.position());
+      LOG_MESSAGE("Team %d: read array", static_cast<int>(myteam.position()));
```
Comment (devreal) [inline dash/test/io/HDF5ArrayTest.cc]: I suggest using `%lu` instead of the cast.

File: dash/test/io/HDF5ArrayTest.cc [hunk 2]
```diff
@@ -568,10 +569,10 @@ TEST_F(HDF5ArrayTest, TeamSplit) {
...
       InputStream is(_filename);
       is >> dio::dataset("array_a") >> array_a;
-      LOG_MESSAGE("Team %d: array read", myteam.position());
+      LOG_MESSAGE("Team %d: array read", static_cast<int>(myteam.position()));
```
Comment (devreal) [inline dash/test/io/HDF5ArrayTest.cc]: I suggest using `%lu` instead of the cast.

File: dash/test/io/HDF5ArrayTest.cc [hunk 3]
```diff
@@ -549,10 +549,11 @@ TEST_F(HDF5ArrayTest, TeamSplit) {
 
       fill_array(array_a, secret);
       myteam.barrier();
-      LOG_MESSAGE("Team %d: write array", myteam.position());
+      LOG_MESSAGE("Team %d: write array", static_cast<int>(myteam.position()));
```
Comment (devreal) [inline dash/test/io/HDF5ArrayTest.cc]: I suggest using `%lu` instead of the cast.

File: dash/test/io/HDF5ArrayTest.cc [hunk 4]
```diff
@@ -536,7 +536,7 @@ TEST_F(HDF5ArrayTest, TeamSplit) {
 
   auto &myteam = team_all.split(num_split);
   LOG_MESSAGE("Splitted team in %d parts, I am %d", num_split,
-              myteam.position());
+              static_cast<int>(myteam.position()));
```
Comment (devreal) [inline dash/test/io/HDF5ArrayTest.cc]: I suggest using `%lu` instead of the cast.

File: dash/test/algorithm/MaxElementTest.cc [hunk 5]
```diff
@@ -40,7 +41,9 @@ TEST_F(MaxElementTest, TestFindArrayDistributeBlockcyclic)
   int block_size   = 7;
   size_t num_units = dash::Team::All().size();
   LOG_MESSAGE("Units: %d, block size: %d, elements: %d",
-              num_units, block_size, _num_elem);
+              static_cast<int>(num_units), block_size,
```
Comment (devreal) [inline dash/test/algorithm/MaxElementTest.cc]: I suggest using `%zu` instead of the cast for `num_units`

**Review by devreal — CHANGES_REQUESTED** [review]
Thanks a lot, I don't know how often I wanted to start doing that as it bothered me as well. Some small points inline.

**Comment by fmoessbauer** [comment]
@devreal your change requests should all be addressed.

**Review by devreal — APPROVED** [review]

**Pull request merged**